### PR TITLE
Minor perf fixes to `get_auth_chain_ids`

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/storage/data_stores/main/event_federation.py [hunk 1]
```diff
@@ -72,13 +72,13 @@ def _get_auth_chain_ids_txn(self, txn, event_ids, include_given):
...
-            chunks = [front_list[x : x + 100] for x in range(0, len(front), 100)]
+            chunks = (front_list[x : x + 100] for x in range(0, len(front), 100))
             for chunk in chunks:
                 clause, args = make_in_list_sql_clause(
                     txn.database_engine, "event_id", chunk
                 )
-                txn.execute(base_sql + clause, list(args))
-                new_front.update([r[0] for r in txn])
+                txn.execute(base_sql + clause, args)
+                new_front.update(r[0] for r in txn)
```
Comment (clokep) [inline synapse/storage/data_stores/main/event_federation.py]: In the past I've used [`flake8-comprehensions`](https://github.com/adamchainz/flake8-comprehensions) to find unnecessary comprehensions like this.

File: synapse/storage/data_stores/main/event_federation.py [hunk 2]
```diff
@@ -72,13 +72,13 @@ def _get_auth_chain_ids_txn(self, txn, event_ids, include_given):
...
             front_list = list(front)
-            chunks = [front_list[x : x + 100] for x in range(0, len(front), 100)]
+            chunks = (front_list[x : x + 100] for x in range(0, len(front), 100))
             for chunk in chunks:
                 clause, args = make_in_list_sql_clause(
                     txn.database_engine, "event_id", chunk
                 )
-                txn.execute(base_sql + clause, list(args))
-                new_front.update([r[0] for r in txn])
+                txn.execute(base_sql + clause, args)
```
Comment (richvdh) [inline synapse/storage/data_stores/main/event_federation.py]: if we're relying on `make_in_list_sql_clause` returning a list, we should probably update its type annotation to say that's what it does?

File: synapse/storage/data_stores/main/event_federation.py [hunk 3]
```diff
@@ -72,13 +72,13 @@ def _get_auth_chain_ids_txn(self, txn, event_ids, include_given):
         while front:
             new_front = set()
             front_list = list(front)
-            chunks = [front_list[x : x + 100] for x in range(0, len(front), 100)]
+            chunks = (front_list[x : x + 100] for x in range(0, len(front), 100))
```
Comment (richvdh) [inline synapse/storage/data_stores/main/event_federation.py]: `batch_iter` is a thing which I think is optimal here?

**Comment by erikjohnston** [comment]
(I merged this manually since github is being slow)

**Comment by clokep** [comment]
(This seems to have been merged in 7b7c3cedf2fdc0d0c05bbc651e0ff5b59921c3a2)

**Pull request closed without merging**